test/Network/ABCI/Test/Types/AppSpec.hs

```diff
@@ -0,0 +1,80 @@
+module Network.ABCI.Test.Types.AppSpec where
+
+import           Network.ABCI.Types.App
+import qualified Network.ABCI.Types.DecodeError as DecodeError
+import qualified Data.Binary.Put as Put
+import qualified Data.ByteString as BS
+import           Data.Bifunctor (first)
+import qualified Data.ByteString.Lazy as LBS
+import           Data.Conduit
+import qualified Data.Conduit.List as CL
+import           Data.Either (isLeft)
+import           Data.Monoid ((<>))
+import           Test.Hspec
+import           Test.QuickCheck
+
+
+main :: IO ()
+main = hspec spec
+
+spec :: Spec
+spec = do
+  describe "decodeLengthPrefix / encodeLengthPrefix" $ do
+
+    -- | This property test (see https://en.wikipedia.org/wiki/QuickCheck)
+    -- for pointers on what this is) checks that these functions are the
+    -- inverse of each other for arbitrarily chunked arbitrary 'ByteString's.
+    -- When the 'Producer' is a real socket, each chunk is the result of a
+    -- 'recv' call. We check for robustness against a peer that decides to
+    -- flush their side of the connection in funky ways.
+
+    it "decoding an encoded bytestring yields the same bytestring" $
+      property $ \(bytelist, nonNegativeChunkSizes) ->
+        let conduit = chunksProducer bytes nonNegativeChunkSizes
+                   .| CL.map (\a -> first DecodeError.print $ decodeLengthPrefix $ encodeLengthPrefix $ [a])
+                   .| consumeValidChunks
+            bytes = BS.pack bytelist
+        in runConduitPure conduit == Right bytes
+
+  describe "decodeLengthPrefix" $ do
+
+    it "fails gracefully when given a string larger than maxMessageLen" $
+      let ginormousSizeVarLen = 8 `BS.cons` runPut (Put.putWord64be maxBound)
+          conduit = yield ginormousSizeVarLen
+                 .| CL.map (first DecodeError.print . decodeLengthPrefix)
+                 .| consumeValidChunks
+      in runConduitPure conduit `shouldSatisfy` isLeft
```

Comment:
why do you need `first DecodeError.print` if you're just checking `isLeft` ?

Reply:
DecodeError has no Show instance